### ISSUE TITLE
Use `tf.where` in attention masking

### ISSUE THREAD

**Issue by jindrahelcl** [ufal/neuralmonkey#534]
Replace the construct 
``weights_all = tf.nn.softmax(energies) * self.attention_mask``

with the one that used the `tf.where` function
